vxfreeswitch/voice.py

```diff
@@ -336,10 +341,17 @@ def register_client(self, client):
 
     def deregister_client(self, client, duration=None):
         client_addr = client.get_address()
+
+        # If originated call has not yet been answered
+        if self._unanswered_channels.get(client_addr):
+            d = self._unanswered_channels[client_addr]
+            d.errback(FreeSwitchClientError('Call is unanswered'))
```

Comment:
I'm worried there's a case here where we `errback` but the deferred is never popped from `_unanswered_channels` if there is no outbound message sent.

If the `client` object is available everywhere, could we store whether the call has been answered as a deferred on the client rather? Then clean-up becomes a lot easier and state is more local?